src/System.Net.Http/src/System/Net/Http/Managed/AuthenticationHandler.cs

```diff
@@ -53,17 +49,26 @@ private bool TrySetBasicAuthToken(HttpRequestMessage request)
 
                 foreach (AuthenticationHeaderValue h in authenticateValues)
                 {
-                    // We only support Basic auth, ignore others
-                    if (h.Scheme == Basic)
+                    // We only support Basic and digest auth, ignore others
+                    if (h.Scheme == AuthenticationHelper.Basic)
                     {
-                        if (!TrySetBasicAuthToken(request))
+                        if (AuthenticationHelper.TrySetBasicAuthToken(request, _credentials))
                         {
+                            response.Dispose();
```

Comment:
We've not yet implemented draining. I'm also not yet sure it's worthwhile, but I guess it can depend on usage.